### PR TITLE
Trace: comm_summary and comm_over_time

### PR DESCRIPTION
**Pull request by hsirkar**
- **comm_summary** computes total message count/volume sent and received per process

image

- **comm_over_time** computes total message count/volume sent over time

image

### CODE REVIEW

File: pipit/trace.py [hunk 1]
```diff
@@ -410,6 +410,57 @@ def message_histogram(self, bins=20, **kwargs):
...
+        """Returns total communication volume per process.
+
+        Returns:
+            pd.DataFrame: DataFrame containing total communication volume sent and
+                received in each process
+        """
+        comm_matrix = self.comm_matrix(output=output)
+
+        # Get total sent and received for each process
+        sent = comm_matrix.sum(axis=1)
```
Comment (ocnkr) [inline pipit/trace.py:509]: I recall a recent discussion about sends and receives being on the incorrect axes. Is this line related to that? Are we certain this is correct?

File: pipit/trace.py [hunk 2]
```diff
@@ -410,6 +410,57 @@ def message_histogram(self, bins=20, **kwargs):
...
+            bins (int, optional): Number of bins in the histogram. Defaults to 50.
+
+        Returns:
+            hist: Volume in each time interval
+            edges: Edges of time intervals
+        """
+        # Filter by send or receive events
+        events = self.events[
+            self.events["Name"].isin(
+                ["MpiSend", "MpiIsend"] if kind == "send" else ["MpiRecv", "MpiIrecv"]
```
Comment (ocnkr) [inline pipit/trace.py]: Is the naming always "MpiSend", or can it occasionally be written as "MPISend"? Do we need to handle that?

Comment (ocnkr) [reply pipit/trace.py]: I guess it should be the same in all the readers so it shouldn't be a problem.

**Comment by bhatele** [comment]
We need to think about why we have an additional "MPISend" event within the "MPI_Send" event and if we can remove it.

**Review by bhatele — APPROVED** [review]
